In [1]:
from __future__ import print_function
import numpy as np
import os
import sys

In [6]:
! curl https://raw.githubusercontent.com/rachelsmc/PUI2018_fb55/master/Lab4_fb55/times.txt > times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0  10636      0 --:--:-- --:--:-- --:--:-- 10718


In [8]:
! mv times.txt $PUIDATA 

mv: cannot stat 'times.txt': No such file or directory


In [9]:
! ls $PUIDATA

setupPUIDATA.ipynb  tapresult.csv  times.txt


In [15]:
times = np.loadtxt("%s/times.txt"%os.getenv("PUIDATA"))

### $H_0$: The commute time on average the same or longer with the new bus route as it was before.

$Average$ $Time_{new}$ - $Average$ $Time_{old}$ $\geq$ 0

### $H_1$: The commute time on average is shorter with the new bus route than it was before.

$Average$ $Time_{new}$ - $Average$ $Time_{old}$ < 0

### Set significance level at 0.05 and Z statistics at 2.

In [20]:
mupop = 36
musample = times.mean()
sigma = 6
n = len(times)

In [23]:
z = (mupop - musample)*(n**0.5)/sigma
z

2.5563971861666701

### Since Z = 2.556 > 2, we reject the null hypothesis at the 0.05 significance level.